Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -86,7 +86,11 @@ struct BinaryDecoder: Decoder {
         fieldEndP = nil
         let start = p
         let c0 = start[0]
-        fieldWireFormat = c0 & 7
+        if let wireFormat = WireFormat(rawValue: c0 & 7) {
```

Comment:
I tried using `FieldTag` here to parse out the field number and wire format, but it measured decidedly slower, even after repeated test runs.  I've not dug into the assembly to figure out what's really going on, so it's possible I was just unlucky when I ran the tests.  Certainly worth further investigation.